### ISSUE TITLE
receptive field

### ISSUE THREAD

**Issue by nnop** [nnop/notes#162]
- [A guide to receptive field arithmetic for Convolutional Neural Networks](https://medium.com/mlreview/a-guide-to-receptive-field-arithmetic-for-convolutional-neural-networks-e0f514068807)
- [Automatically Compute Receptive Field in CNN](http://kukuruza.github.io/receptive_field)
- [【深度学习】一张图看懂Receptive Field](https://medium.com/@sunnerli/simple-introduction-about-hourglass-like-model-11ee7c30138)

**Comment by nnop** [comment]
- [Gradient based receptive field estimation](https://github.com/fornaxai/receptivefield)
- [Tensorflow receptive field](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/receptive_field)
- [Recptive field calculator](http://fomoro.com/tools/receptive-fields/index.html)

**Comment by nnop** [comment]
**papers**

- Understanding the Effective Receptive Field in Deep Convolutional Neural Networks
  - [NIPS page](https://papers.nips.cc/paper/6203-understanding-the-effective-receptive-field-in-deep-convolutional-neural-networks)
  - blog: [1](http://blog.christianperone.com/2017/11/the-effective-receptive-field-on-cnns/), [2](https://wiki.math.uwaterloo.ca/statwiki/index.php?title=Understanding_the_Effective_Receptive_Field_in_Deep_Convolutional_Neural_Networks)
- Structured Receptive Fields in CNNs
- Receptive Field Block Net for Accurate and Fast Object Detection
- Enlarging Effective Receptive Field of Convolutional Neural Networks for Better Semantic Segmentation
- Dilated Residual Networks
- [**Bolei Zhou**](http://people.csail.mit.edu/bzhou/)'s works

**Comment by nnop** [comment]
**coordinates mapping between blobs**

caffe issues

- Should pooling regions be identical to convolution regions? # 1318
- Augment layers with their induced coordinate maps # 1637
  -  Augment layers with their induced coordinate maps # 1975
      - **Python/net spec coordinate map and crop computation** # 3613
  - Crop layer for automatically aligning computations # 1639
      - Crop layer for automatically aligning computations # 1976
          - ND Crop layer # 3570


Kaiming's lecture at ICCV 2015 tutorial:
![image](https://user-images.githubusercontent.com/23091984/38171659-0305ebf2-35d1-11e8-9a89-ffade4967122.png)
